### DISCUSSION TITLE
Delaying DataGrid Filtering with ReadData

### DISCUSSION THREAD

**Discussion by christophwille** [Megabit/Blazorise#2561] [Q&A]
I have server-side paging implemented, that works great. Now I added filtering via Filterable=true on the respective columns. 

Problem: the moment I type a single character, my OnReadData is being called. 

Is there a way (similar to what many AutoComplete boxes do) to delay the call to see "oh, does the user keep typing?" (minimizing pointless server calls) Is there a best practice?

Chris

**Comment by stsrki** [comment]
You could make use of the TextEdit delay feature. For that, you will need to add a FilterTemplate.

**Example**

```cshtml
<DataGridColumn TItem="Employee" Field="@nameof( Employee.FirstName )" Caption="First Name" Validator="@CheckFirstName" Editable="true">
    <FilterTemplate>
        <TextEdit DelayTextOnKeyPress="true" DelayTextOnKeyPressInterval="1000" Placeholder="Search name" Text="@context.SearchValue?.ToString()" TextChanged="@(v=> context.TriggerFilterChange(v))" />
    </FilterTemplate>
</DataGridColumn>
```

**Reply by christophwille** [reply]
Thanks, yes, that does the trick of giving enough delay to allow the user to type and then do the actual searching.

**Reply by David-Moreira** [reply]
Yea and that can also be configured globally on the startup, right strski? So there's more room for you to play with those configurations there.

**Reply by stsrki** [reply]
It can be configured globally but in that case everything will be delayed. It's OK if that is what you want. It all depends on the project requirements.

**Reply by David-Moreira** [reply]
Yep. Just saying there's more options to play with it here. 👍
Also this issue https://github.com/Megabit/Blazorise/issues/1926 
Is kinda related if you want to encapsulate the new filter or other column configurations in a single component.
